dev/source/docs/git-branch.rst

```diff
@@ -118,3 +118,43 @@ branch and commit it.
       :target: ../_images/GitHubForWindowsClient_Publish_Push.png
 
       GitHub for Windows Client: Pushing changes
+
+
+Updating a branch with upstream changes
+---------------------------------------
+
+Sometimes, your branch will get out date with master, or you will have merge conflicts to fix. 
+
+`sync your fork with upstream <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork>`__.
+
+Note: ArduPilot does not use a merge-strategy; instead it uses rebase.
+
+To add ArduPilot as remote named ``upstream``:
+
+   ::
+
+       git remote add upstream https://github.com/ArduPilot/ardupilot.git
+       git remote -v
+       > origin    https://github.com/YOUR_USERNAME/ardupilot.git (fetch)
+       > origin    https://github.com/YOUR_USERNAME/ardupilot.git (push)
+       > upstream  https://github.com/ArduPilot/ardupilot.git (fetch)
+       > upstream  https://github.com/ArduPilot/ardupilot.git (push)
+
+
+To sync your ``master`` branch with ``upstream``:
+
+   ::
+
+       git checkout master
+       git fetch upstream
+       git rebase upstream/master
+       git push origin
+
+To get your ``feature`` branch up to date with ``master``:
+
+   ::
+
+       git checkout ardupilot_git_tutorial
```

Comment:
I usually make a backup of the branch after the checkout if I'm uncertain it would be an easy rebase for me.  Something like 'git branch backup/ardupilot_git_tutorial-202312181501`

Saves losing work when you stuff the rebase up.

Reply:
Isn't that with git rebase --abort is for? 
Rebase ensures you have everything comitted.

Reply:
Depends whether you realise you've stuffed up befpre or after finishing the rebase...